backend/inspirehep/records/api/mixins.py

```diff
@@ -137,9 +147,13 @@ def files(self):
         bucket = self.get_bucket()
         return self.files_iter_cls(self, bucket=bucket, file_cls=self.file_cls)
 
+    def delete_all_files(self):
+        for key in list(self.files.keys):
+            del self.files[key]
```

Comment:
I'm testing this locally now. When you do this, file is marked as removed and when you are trying to add it, it's pulled from local file object. Which is removed already. 
You should modify `get_file_object` method to query only for non deleted files.

Reply:
I still believe that you should fix `get_file_object` method to query only for non deleted files.

Reply:
I completely removed the method :/